### ISSUE TITLE
Conditions run twice

### ISSUE THREAD

**Issue by Niromash** [ItalyPaleAle/svelte-spa-router#236]
Hi, I have a problem, my conditions run twice.
For example I made a function as basic as 
![image](https://user-images.githubusercontent.com/27890221/129188157-b3f91027-8c16-49b8-9efc-2fca5e202e17.png)
but in my console, it logged the message twice 
![image](https://user-images.githubusercontent.com/27890221/129188074-386b5521-d592-4100-9c25-97b33be0393e.png)
and for web requests, I try to avoid having a duplicate request made for nothing
Thanks

**Comment by ItalyPaleAle** [comment]
I will need more information please. Can you share your code? Or at least a "MVP" that can allow reproducing this?

**Comment by Niromash** [comment]
yes sorry, I made my project public https://github.com/Niromash/website

**Comment by ItalyPaleAle** [comment]
Thanks for sharing the code, i found the issue.

This line in `src/App.svelte`:

```svelte
{#key darkMode && $location}
```

Because the `<Router>` component is shown inside a keyed block based on `$location`, when the `$location` changes, Svelte re-renders what's inside the keyed block. This causes the Router to be un-mounted and re-mounted, and the trigger for a location change is fired twice. In fact, not just the route pre-conditions run twice, but every event (routeLoaded and routeLoading too).

Please remove `$location` from a keyed block :)

**Comment by Niromash** [comment]
yeah I know, but I had this to put a transition on all the routes automatically without putting the transition on each route, is there a solution to do it without making the problem I had?

**Comment by ItalyPaleAle** [comment]
I would recommend putting the transition on each route

**Comment by Niromash** [comment]
Ok thanks, have a nice day
